library/src/main/java/com/google/android/exoplayer/metadata/id3/Id3Parser.java

```diff
@@ -107,7 +107,40 @@ public boolean canParse(String mimeType) {
         System.arraycopy(frame, descriptionEndIndex + delimiterLength(encoding), objectData, 0,
             objectDataSize);
         id3Frames.add(new GeobFrame(mimeType, filename, description, objectData));
-      } else {
+      } else if (frameId0 == 'A' && frameId1 == 'P' && frameId2 == 'I' && frameId3 == 'C') {
+        // Check frame ID == APIC
+        int encoding = id3Data.readUnsignedByte();
+        String charset = getCharsetName(encoding);
+
+        byte[] frame = new byte[frameSize - 1];
+        id3Data.readBytes(frame, 0, frameSize - 1);
+
+        int firstZeroIndex = indexOf(frame, 0, (byte) 0);
+        String mimeType = new String(frame, 0, firstZeroIndex, "ISO-8859-1");
+        byte pictureType = frame[firstZeroIndex+1];
+
+        int descriptionStartIndex = firstZeroIndex + 2;
+        int descriptionEndIndex = indexOf(frame, descriptionStartIndex, (byte) 0);//indexOfEOS(frame, descriptionStartIndex, encoding);
```

Comment:
No. The last parameter of indexOf takes the byte you are looking for (in our case, the "0" byte that separates the description from the binary data after it). indexOfEOS takes the encoding.